jupyter_server/services/auth/manager.py

```diff
@@ -0,0 +1,53 @@
+"""An AuthorizationManager for use in the Jupyter server.
+
+The default manager is a No-op manager, NOPAuthorizationManager
+"""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+from traitlets.config import LoggingConfigurable
+
+
+class AuthorizationManager(LoggingConfigurable):
+    """Base class for managing authorization to resources
+    in the Jupyter Server.
+
+    All authorization managers used in Jupyter Server
+    should inherit from this base class and, at the very minimum,
+    override and implement an `is_authorized` method with the
+    same signature as in this base class.
+
+    The `is_authorized` method is called by the `@authorized` decorator
+    in JupyterHandler. If it returns True, the incoming request
+    to the server is accepted; if it returns False, the server
+    returns a 401 (Not Authorized) error code.
+    """
+
+    def is_authorized(self, handler, subject, action, resource):
```

Comment:
Should we add type hints?  I find them quite helpful, especially as documentation.  Not sure if `handler: JupyterHandler` would trigger a circular dependency though.